src/components/RequestPanel/RequestPanel.js

```diff
@@ -0,0 +1,291 @@
+      }
+    },
+    [setProgress]
+  )
+
+  // Get request statuses
+  const requestStatus = useMemo(() => {
+    if (!requests) {
+      return []
+    }
+
+    const requiredConfirmedRequests = requests
+      .map((req, index) => ({ ...req, index }))
+      .filter(req => req.ensureConfirmation)
+
+    const { confirmed, errorProcessing, failed, processed } = progress
+    const status = (request, index) => {
+      // Request processing failed
+      if (errorProcessing !== -1 && index >= errorProcessing) {
+        return REQUEST_STATUS_PROCESSING_FAILED
+      }
+
+      // Request confirmation failed
+      if (failed !== -1 && index >= failed) {
+        return REQUEST_STATUS_FAILED
+      }
+
+      if (request.ensureConfirmation) {
+        const confirmedIndex = requiredConfirmedRequests.findIndex(
+          req => req.index === index
+        )
+
+        // Request confirmed
+        if (confirmedIndex < confirmed) {
+          return REQUEST_STATUS_CONFIRMED
+        }
+
+        // Request pending
+        if (index < processed) {
+          return REQUEST_STATUS_PENDING
+        }
+      }
+
+      // Request processed
+      if (index < processed) {
+        return REQUEST_STATUS_PROCESSED
+      }
+
+      // Transaction not processed
+      return REQUEST_STATUS_UPCOMING
+    }
+
+    return requests.map((request, index) => status(request, index))
+  }, [progress, requests])
+
+  const allSuccess = useMemo(() => {
+    const { confirmed, processed } = progress
+
+    const requiredConfirmedRequests = requests.filter(
+      request => request.ensureConfirmation
+    ).length
+
+    const allProcessed = processed > 0 && processed === requests.length
+    const allConfirmed = confirmed === requiredConfirmedRequests
+    return allProcessed && allConfirmed
+  }, [progress, requests])
+
+  const maxAttemptsReached = attempts >= MAX_ATTEMPTS
+
+  // Create requests
+  useEffect(
+    () => {
+      if (!progress.requesting || maxAttemptsReached) {
+        return
+      }
+
+      if (attempts === 0) {
+        setProgress(({ requesting }) => ({
+          ...INITIAL_STATE,
+          requesting,
+        }))
+      } else {
+        setProgress(progress => ({
+          ...progress,
+          errorProcessing: -1,
+          failed: -1,
+        }))
+      }
+
+      let cancelled = false
+
+      const createRequests = async () => {
+        const filteredRequests = requests.slice(progress.processed)
+
+        for (const request of filteredRequests) {
+          if (cancelled) {
+            break
+          }
+
+          try {
+            const requestProcess = await processRequest(request)
+
+            if (!cancelled) {
+              if (request.isTx && request.ensureConfirmation) {
+                await ensureRequestConfirmed(requestProcess)
+
+                // Execute callback function if required
+                if (typeof request?.callback === 'function') {
+                  request.callback()
```

Comment:
Will these callback functions ever change a component's state?

If not, I think we should still run them even if we've unmounted (but obviously not run any new ones).

Reply:
They are not supposed to change component state no. 
They are meant as functions to execute after we ensure that the tx has been confirmed e.g saving one time code to local storage.

From convo offline we decided to remove the cancel condition here